modules/renter/hostdb/scan.go

```diff
@@ -202,32 +202,20 @@ func (hdb *HostDB) threadedScan() {
 				}
 			}
 
-			// Randomize the slice by swapping each element with an element
-			// that hasn't been visited yet.
-			for i := 0; i < len(random); i++ {
-				N, err := rand.Int(rand.Reader, big.NewInt(int64(len(random)-i)))
-				if err != nil {
-					if build.DEBUG {
-						panic(err)
-					}
-				} else {
-					break
-				}
-
-				n := int(N.Int64()) + i
-				tmp := random[i]
-				random[i] = random[n]
-				random[n] = tmp
-			}
-
-			// Select the first InactiveHostCheckupQuantity hosts from the
-			// shuffled list and scan them.
+			// Generate a random ordering of up to InactiveHostCheckupQuantity
+			// hosts.
 			n := InactiveHostCheckupQuantity
-			if len(random) < InactiveHostCheckupQuantity {
-				n = len(random)
+			if n > len(entries) {
+				n = len(entries)
 			}
-			for i := 0; i < n; i++ {
-				hdb.scanHostEntry(random[i])
+			perm, err := crypto.Perm(n)
```

Comment:
I don't like the name `perm`, makes me think of permanent. I think that the full name `permutation` is justified, simply due to the namespace collision.